pkg/alert/alert.go

```diff
@@ -0,0 +1,534 @@
+		alert.log.Info("no new failed tests found, nothing to post in slack")
+		return nil
+	}
+	message := createAlertMessage(failedTests, alert)
+	splitedMessage := splitSlackMessage(message, 3900)
+	messagePrefix := "*🔥 New Testmachinery Alerts:* \n"
+	for i, messageSplitItem := range splitedMessage {
+		if i != 0 {
+			messagePrefix = ""
+		}
+		if err := client.PostMessage(channel, fmt.Sprintf("%s```%s```", messagePrefix, messageSplitItem)); err != nil {
+			return err
+		}
+		time.Sleep(1200 * time.Millisecond) // need to wait 1 sec due to slack limits
+	}
+	alert.log.Info("Sent slack alert message", "failing tests", len(failedTests))
+	if err := alert.filePostedAlerts(failedTests); err != nil {
+		return err
+	}
+	return nil
+}
+
+//createAlertMessage creates the alert message
+func createAlertMessage(failedTests map[string]TestDetails, alert *Alert) string {
+	sortedKeys := make([]string, 0, len(failedTests))
+	for k := range failedTests {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	writer := &strings.Builder{}
+	table := tablewriter.NewWriter(writer)
+	header := []string{"Test", "Landscape", "Provider", "K8s Ver", "OS", "Success", "Alert Reason", "Last failure"}
+
+	content := make([][]string, 0)
+	row := 0
+	for _, mapKey := range sortedKeys {
+		test := failedTests[mapKey]
+		failureReason := ""
+		if test.SuccessRate < float64(alert.cfg.SuccessRateThresholdPercent) {
+			failureReason = fmt.Sprintf("success rate < %d%%", alert.cfg.SuccessRateThresholdPercent)
+		} else if test.FailedContinuously {
+			failureReason = fmt.Sprintf(">%d failures in row", alert.cfg.ContinuousFailureThreshold-1)
+		}
+		newRow := []string{test.Name,
+			test.Landscape,
+			util.StringDefault(test.Cloudprovider, "-"),
+			util.StringDefault(test.K8sVersion, "-"),
+			util.StringDefault(test.OperatingSystem, "-"),
+			fmt.Sprintf("%d%%", int(test.SuccessRate)),
+			failureReason,
+			test.LastFailedTimestamp,
+		}
+		content = append(content, newRow)
+		row++
+	}
+
+	table.SetHeader(header)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.AppendBulk(content)
+	table.Render()
+	return writer.String()
+}
+
+//PostRecoverMessageToSlack posts alerts to slack
+func (alert *Alert) PostRecoverMessageToSlack(client slack.Client, channel string, recoveredTests map[string]TestDetails) error {
+	if len(recoveredTests) == 0 {
+		alert.log.Info("no new recovered tests found, nothing to post in slack")
+		return nil
+	}
+	message := createRecoverMessage(recoveredTests)
+	splittedMessage := splitSlackMessage(message, 3900)
+	messagePrefix := "*🍏 Testmachinery Tests Got Healthy:* \n"
+	for i, messageSplitItem := range splittedMessage {
+		if i != 0 {
+			messagePrefix = ""
+		}
+		if err := client.PostMessage(channel, fmt.Sprintf("%s```%s```", messagePrefix, messageSplitItem)); err != nil {
+			return errors.Wrap(err, "failed to post a slack message")
+		}
+		time.Sleep(1200 * time.Millisecond) // need to wait 1 sec due to slack limits
```

Comment:
`time.Second` ?

Reply:
I would like wait a little more than a second to avoid any possible throtteling